### PR TITLE
Add subscription filter to log group

### PR DESCRIPTION
**Pull request by ejayesh**
This PR is to add a subscription filter with a custom CF yaml file to the application log group.  This is added to the service addon.

### CODE REVIEW

File: dbt_copilot_helper/templates/addons/svc/subscription-filter.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# SubscriptionFilter requires the Destination ARN of the CloudWatch Logs in the main archive logs account.
```
Comment (yusufsheiqh) [inline dbt_copilot_helper/templates/addons/svc/subscription-filter.yml:1]: Looks to be missing the `version_info`.

```suggestion
# {% version_info %}
# SubscriptionFilter requires the Destination ARN of the CloudWatch Logs in the main archive logs account.

```

File: tests/copilot_helper/fixtures/make_addons/expected/web/addons/subscription-filter.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# SubscriptionFilter requires the Destination ARN of the CloudWatch Logs in the main archive logs account.
```
Comment (yusufsheiqh) [inline tests/copilot_helper/fixtures/make_addons/expected/web/addons/subscription-filter.yml:1]: To accommodate the above change for tests.

```suggestion
# {% version_info %}
# SubscriptionFilter requires the Destination ARN of the CloudWatch Logs in the main archive logs account.

```

File: tests/copilot_helper/fixtures/make_addons/expected/web/addons/subscription-filter.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    Description: The name of the service, job, or workflow being deployed.
+Resources:
+  SubscriptionFilter:
+    Type: AWS::Logs::SubscriptionFilter
+    Properties:
+      RoleArn: !Sub "arn:aws:iam::${AWS::AccountId}:role/CWLtoSubscriptionFilterRole"
+      LogGroupName: !Sub "/copilot/${App}/${Env}/${Name}"
+      FilterName: !Sub "${App}-${Env}-stream"
+      FilterPattern: ""
+      DestinationArn: "arn:aws:logs:eu-west-2:812359060647:destination:cwl_log_destination"
```
Comment (yusufsheiqh) [inline tests/copilot_helper/fixtures/make_addons/expected/web/addons/subscription-filter.yml:20]: Do we need to add a comment to say this account number corresponds to the Log Archive AWS account explicitly? Maybe bringing the comment at the top to here instead(?).

Comment (ejayesh) [reply tests/copilot_helper/fixtures/make_addons/expected/web/addons/subscription-filter.yml:20]: yes I did think about putting it there, you are right I will move it

File: dbt_copilot_helper/templates/addons/svc/subscription-filter.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+Resources:
+  SubscriptionFilter:
+    Type: AWS::Logs::SubscriptionFilter
+    Properties:
+      RoleArn: !Sub "arn:aws:iam::${AWS::AccountId}:role/CWLtoSubscriptionFilterRole"
+      LogGroupName: !Sub "/copilot/${App}/${Env}/${Name}"
+      FilterName: !Sub "${App}-${Env}-stream"
+      FilterPattern: ""
+      # SubscriptionFilter requires the Destination ARN of the CloudWatch Logs in the main archive logs account.
+      DestinationArn: "arn:aws:logs:eu-west-2:812359060647:destination:cwl_log_destination"
```
[the remark this answers is not in the document]
Comment (yusufsheiqh) [reply dbt_copilot_helper/templates/addons/svc/subscription-filter.yml:21]: We might be able to add conditions. I think we can do that in CFN templates.

[the remark this answers is not in the document]
Comment (ejayesh) [reply dbt_copilot_helper/templates/addons/svc/subscription-filter.yml:21]: I will create a new ticket for this

[the remark this answers is not in the document]
Comment (ejayesh) [reply dbt_copilot_helper/templates/addons/svc/subscription-filter.yml:21]: ticket created

Comment (WillGibson) [inline dbt_copilot_helper/templates/addons/svc/subscription-filter.yml:21]: I'm wondering if we can parameterise the account number so it's not part of the library?

Comment (ejayesh) [reply dbt_copilot_helper/templates/addons/svc/subscription-filter.yml:21]: we have the same issue with the appconfig-ipfilter.yml.  we have the arn in the code.  Maybe something we pick up seperately for all files

Comment (WillGibson) [reply dbt_copilot_helper/templates/addons/svc/subscription-filter.yml:21]: OK, happy to approve this if you can make a ticket to look at solving that.

**Review by WillGibson — APPROVED** [review]

**Pull request merged**